### PR TITLE
Added new gcip.addons.security.sops module.

### PR DESCRIPTION
**Pull request by dvonessen**
Added sops_export_decrypted_values function to sops module.

What does this PR change? What problem does it solve?
-----------------------------------------------------

Added new gcip.addons.security.sops module.

Was the change discussed in an issue?
------------------------------------------------------------

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CONTRIBUTING.md)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [x] I have added api documentation for the changes
- [x] I added a changelog entry to the unreleased section with the PR number. [Changelog](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CHANGELOG.md)
- [x] Linting is done, and linting jobs are successfull
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: gcip/addons/security/sops.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+__author__ = "Daniel von Eßen"
+__copyright__ = "Copyright 2020 DB Systel GmbH"
+__credits__ = ["Daniel von Eßen", "Thomas Steinbach"]
+# SPDX-License-Identifier: Apache-2.0
+__license__ = "Apache-2.0"
+__maintainer__ = "Thomas Steinbach"
+__email__ = "daniel.von-essen@deutschebahn.com"
+
+
+def sops_export_decrypted_values(path: str) -> str:
```
Comment (ThomasSteinbach) [inline gcip/addons/security/sops.py:10]: I think it is more secure and "sops native" to write a function that will take the path of the sops file to encrypt as well as a command to execute and wrapt that command into the `sops exec-env` command:
https://github.com/mozilla/sops/blob/master/README.rst#218passing-secrets-to-other-processes

Comment (ThomasSteinbach) [reply gcip/addons/security/sops.py:10]: sth. like

```
def sops_exec_env(path: str, command: str) -> str:
    return f"sops exec-env {path} '{command}‘"
```

Comment (dvonessen) [reply gcip/addons/security/sops.py:10]: That would be awsome. I think that could be a second function. 
My intention is, to export variables in predefined jobs. E.g. DockerClientConfig needs `REGISTRY_USERNAME` and `REGISTRY_PASSWORD` env vars.
`DockerClientConfig` is used in preconfigured jobs like `crane.push` or `crane.pull`. 
To decrypt and export env vars, I want to just have to `decrypt` and `exoprt` those variables. 

_sops/secrets.env_
```env
REGISTRY_USERNAME=myusername
REGISTRY_PASSWORD=secretpassword
``` 
_gitlab-ci.py_

```python
pipeline = Pipeline()
crane_job = crane.push("my_registry_url", "image", "v1.1.1") # crane_job uses default DockerClientConfig
crane_job.prepend_scripts(sops_export_decrypted_values("sops/secrets.env"))
pipeline.add_children(crane_job)
pipeline.write()  
```
_generated_pipeline.yaml_
```yaml
job:
  script:
    - export $(sops -d sops/secrets.env)
    - mkdir -p $HOME/.docker
    - 'echo "{\"auths\": {\"index.docker.com\": {\"username\": \"$REGISTRY_USERNAME\",
  \"password\": \"$REGISTRY_PASSWORD\"}}}" > $HOME/.docker/config.json'
    - crane push my_registry_url/image:v1.1.1
```

Comment (ThomasSteinbach) [reply gcip/addons/security/sops.py:10]: check

**Review by ThomasSteinbach — APPROVED** [review]

**Pull request merged**